cookbook/integrations/kubernetes/kftensorflow/tf_mnist.py

```diff
@@ -0,0 +1,255 @@
+    checkpoint_prefix = os.path.join(checkpoint_dir, "ckpt_{epoch}")
+
+    # define a callback for printing the learning rate at the end of each epoch
+    class PrintLR(tf.keras.callbacks.Callback):
+        def on_epoch_end(self, epoch, logs=None):
+            print(
+                "\nLearning rate for epoch {} is {}".format(
+                    epoch + 1, model.optimizer.lr.numpy()
+                )
+            )
+
+    # put all the callbacks together
+    callbacks = [
+        tf.keras.callbacks.TensorBoard(log_dir="./logs"),
+        tf.keras.callbacks.ModelCheckpoint(
+            filepath=checkpoint_prefix, save_weights_only=True
+        ),
+        tf.keras.callbacks.LearningRateScheduler(decay),
+        PrintLR(),
+    ]
+
+    # train the model
+    model.fit(train_dataset, epochs=hyperparameters.epochs, callbacks=callbacks)
+
+    # save the model
+    model.save(MODEL_FILE_PATH, save_format="tf")
+
+    return model, checkpoint_dir
+
+
+# %%
+# Evaluation
+# ==========
+#
+# We define ``test_model`` to evaluate loss and accuracy on the test dataset.
+def test_model(
+    model: tf.keras.Model, checkpoint_dir: str, eval_dataset: tf.data.Dataset
+) -> Tuple[float, float]:
+    model.load_weights(tf.train.latest_checkpoint(checkpoint_dir))
+
+    eval_loss, eval_acc = model.evaluate(eval_dataset)
+
+    return eval_loss, eval_acc
+
+
+# %%
+# Defining an MNIST TensorFlow Task
+# ==================================
+#
+# We initialize compute requirements and task output signature. Next, we define a ``mnist_tensorflow_job`` to kick off the training and evaluation process.
+# The task is initialized with ``TFJob`` with certain values set:
+#
+# * ``num_workers``: integer determining the number of worker replicas to be spawned in the cluster for this job
+# * ``num_ps_replicas``: number of parameter server replicas to use
+# * ``num_chief_replicas``: number of chief replicas to use
+#
+# MirroredStrategy uses an all-reduce algorithm to communicate the variable updates across the devices.
+# Hence, ``num_ps_replicas`` is not useful in our example.
+#
+# .. note::
+#   If you'd like to understand the various Tensorflow strategies in distributed training, refer to the `Types of strategies <https://www.tensorflow.org/guide/distributed_training#types_of_strategies>`__ section in the TensorFlow documentation.
+training_outputs = NamedTuple(
+    "TrainingOutputs", accuracy=float, loss=float, model_state=FlyteDirectory
+)
+
+if os.getenv("SANDBOX") != "":
+    mem = "1000Mi"
+    gpu = "0"
+    ephemeral_storage = "500Mi"
+    storage = "500Mi"
+else:
+    mem = "10Gi"
+    gpu = "2"
+    ephemeral_storage = "500Mi"
+    storage = "10Gi"
+
+
+@task(
+    task_config=TfJob(num_workers=2, num_ps_replicas=1, num_chief_replicas=1),
+    retries=2,
+    cache=True,
+    cache_version="1.0",
+    requests=Resources(
+        gpu=gpu,
+        mem=mem,
+        storage=storage,
+        ephemeral_storage=ephemeral_storage,
+    ),
```

Comment:
nit: define the `Resources` object in the `os.getenv("SANDBOX")` conditional above so it can be reused in the `requests` and `limits` argument